### ISSUE TITLE
Build failed for Release (2.0) x64 platform

### ISSUE THREAD

**Issue by jimfoltz** [SketchUp/ruby-c-extension-examples#6]
Using VS Express 2015 for Windows Desktop.

I had to change the _Runtime Library_ to use the `/MD` option instead of the `/MT` option in order to build `SUEXT_HelloWorld`  for Release (2.0) on the x64 Platform. 

`Properties > C/C++ > Code Generation >Runtime Library`

![2016-09-10_130113](https://cloud.githubusercontent.com/assets/137514/18412308/4aafebc0-7759-11e6-8be6-9bee38e35353.png)

**Comment by thomthom** [comment]
Which compiler did you use? Did you upgrade the project to use the VS2015 compiler?
If you do that there are changes that needs to be made to the Ruby header as otherwise you get conflicts between Ruby and CRT functions.

**Comment by jimfoltz** [comment]
Using VS Express 2015 for Windows Desktop.

> Did you upgrade the project to use the VS2015 compiler?
I just double-clicked the `.sln` and tried to  build the `SUEXT_HelloWorld` project. I never saw anything about upgrading. VS2015 is the only version I have installed on this machine. 

The Debug (2.0) x64 build compiled and worked in SketchUp. The Release(2.0) x64 failed until I made the above changes. Then it compiled and also worked in SketchUp.

**Comment by thomthom** [comment]
> Using VS Express 2015 for Windows Desktop.
The IDE version is not necessarily the compiler version. You can compile with the 2010 (or any other older compiler) compiler using VS2015.

hm... was Debug set to MDd or MTd?
MD will make you have to ensure the user have the VS CRT installed in order to be able to load the extension.

**Comment by jimfoltz** [comment]
Debug is set to `/MDd` - that is where I got the idea to try `/MD` for the release version also.  When I did revert back to `/MT` and commented out the conflicting macros in the file `missing.h` , it built a `.so` file instead of a `.dll` file.

I have only VS2015 installed on this machine so I assume it is using whatever version of compiler and libraries came with the installer.

**Comment by thomthom** [comment]
Yea, if you don't have any older compiler versions it can only compile with VS2015.

When you switch between MD and MT the CRT one have to adjust some macros to prevent Ruby from defining global symbols that would otherwise conflict.

I think I set Release to MT because that's when you build for distribution - and to make things easier I set that to avoid the C Extension needing the CRT redistrobutable.

For debug MDd is fine - as it will most likely only be run on your machine.

**Comment by thomthom** [comment]
This seems to work with MT in release: (I copied from SUbD)

**RubyLib.h**
```
#ifndef SU_UTILS_RUBYLIB_H_
#define SU_UTILS_RUBYLIB_H_

// Must disable the min/max macros defined by windows.h to avoid conflict with
// std::max and std:min. Ruby includes windows.h so it must be disabled here.
// http://stackoverflow.com/a/2789509/486990
#define NOMINMAX

// Needed for VS2015.
#if _MSC_VER >= 1900
#define HAVE_STRUCT_TIMESPEC 1
// Ruby was configured for Visual Studio 2010. These macros are needed to
// prevent compiler errors with Visual Studio 2013.
#define HAVE_ACOSH 1
#define HAVE_CBRT 1
#define HAVE_ERF 1
#define HAVE_TGAMMA 1
#define HAVE_ROUND 1
#endif

#include <ruby.h>
#ifdef HAVE_RUBY_ENCODING_H
#include <ruby/encoding.h>
#endif

#endif // SU_UTILS_RUBYLIB_H_
```

**Comment by jimfoltz** [comment]
Thanks Thom.  I started down the opposite route commenting out the conflicting defines in the Ruby header which worked also. You're way is cleaner and also favors the Ruby version of the functions which is, I assume, the referred version in this case?

**Comment by thomthom** [comment]
Yea, I also edited the Ruby headers at first. But I found that as I made other extensions I had to do the same thing over and over. And declaring the defines makes it easier to set up configs for different compilers.

Note that in my example you might need some of these defines if you were using VS2013 as well - so in such an environment you might want to add another version check there.

It's on our list to update the examples - a project we have already started internally.
